manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/api/internal/ShardAPI.java

```diff
@@ -127,11 +130,29 @@ public Response getProcessors() {
             @APIResponse(description = "Forbidden.", responseCode = "403"),
             @APIResponse(description = "Internal error.", responseCode = "500", content = @Content(mediaType = MediaType.APPLICATION_JSON))
     })
-    @Operation(hidden = true, summary = "Update a Processor status.", description = "Update a Processor status.")
+    @Operation(hidden = true, summary = "Update Processors' status.", description = "Update Processors' status.")
     @PUT
     @Path("processors")
-    public Response updateProcessorStatus() {
-        return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), "Not implemented yet.").build();
+    public Response updateProcessorsStatus(List<ProcessorStatusDTO> statusDTOs) {
+        String shardId = identityResolver.resolve(jwt);
+        failIfNotAuthorized(shardId);
+
+        statusDTOs.forEach(this::updateProcessorStatus);
```

Comment:
Might be the case to be a bit more "defensive" here: if an update throws an exception we skip all the next updates in the list (and in case it is not a transient failure for any reasons we keep failing in all the next calls from the shard). A try/catch at this level (with a warn log) might be a good idea here